### PR TITLE
reference/sql: list valid length value ranges in VARCHAR for different charsets

### PR DESCRIPTION
**Pull request by tangenta**
### What is changed, added or deleted? 

The current [description](https://github.com/pingcap/docs-cn/blob/master/v3.1/reference/sql/data-types/string.md) of string type is confusing. 
> VARCHAR 的最大实际长度由最长的行的大小和使用的字符集确定。

A few sentences are clarified.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [ ] This PR is to align with:
- [x] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [x] All active versions: dev, v3.0, v2.1, v3.1
- [ ] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

### CODE REVIEW

File: v3.1/reference/sql/data-types/string.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ TiDB 支持 MySQL 所有的字符串类型，包括 `CHAR`、`VARCHAR`、`BINARY
 
 ### `CHAR` 类型
 
-定长字符串。`CHAR` 列的长度固定为创建表时声明的长度。长度可以为从 0 到 255 的任何值。当保存 CHAR 值时，在它们的右边填充空格以达到指定的长度。
+定长字符串。`CHAR` 列的长度固定为创建表时声明的长度。在 ASCII 字符集下，长度可以为从 0 到 255 的任何值。当保存 CHAR 值时，在它们的右边填充空格以达到指定的长度。
```
Comment (djshow832) [inline v3.1/reference/sql/data-types/string.md]:
```suggestion
定长字符串。`CHAR` 列的长度固定为创建表时声明的长度。在 `ASCII` 字符集下，长度可以为从 0 到 255 的任何值。当保存 CHAR 值时，在它们的右边填充空格以达到指定的长度。
```

Comment (djshow832) [reply v3.1/reference/sql/data-types/string.md]: Same for other `ASCII` in this file.

**Comment by yikeke** [comment]
After this PR merges, sre-bot will automatically cherry-pick this PR to the `release-3.1`, `release-3.0`, and `release-2.1` branches (the needs-cherry-pick-xxx labels will trigger the behavior). FYI. @tangenta

**Comment by yikeke** [comment]
ping @djshow832 Please approve this PR if you have no further comments.

**Comment by TomShawn** [comment]
@djshow832 PTAL, thanks!

File: reference/sql/data-types/string.md [hunk 2]
```diff
@@ -21,7 +21,17 @@ TiDB 支持 MySQL 所有的字符串类型，包括 `CHAR`、`VARCHAR`、`BINARY
 
 ### `VARCHAR` 类型
 
-变长字符串。M 表示最大列长度，范围是 0 到 65535。VARCHAR 的最大实际长度由最长的行的大小和使用的字符集确定。
+变长字符串。M 表示最大列长度（字符的最大个数）。`VARCHAR` 的空间占用大小不得超过 65535 字节。在选择 `VARCHAR` 长度时，应当根据最长的行的大小和使用的字符集确定。
+
+对于不同的字符集，单个字符所占用的空间可能有所不同。以下表格是各个字符集下在单个字符占用的字节数，以及 `VARCHAR` 列长度的取值范围：
```
Comment (djshow832) [inline reference/sql/data-types/string.md]:
```suggestion
对于不同的字符集，单个字符所占用的空间可能有所不同。以下表格是各个字符集下单个字符占用的字节数，以及 `VARCHAR` 列长度的取值范围：
```

**Comment by yikeke** [comment]
PTAL again. @djshow832

**Comment by lilin90** [comment]
Hi @sumens, I see your comment https://github.com/pingcap/docs-cn/pull/2245#issuecomment-603266859 in another PR. That one has been updated in the English version. Would you like to help translate this one?

**Comment by sumens** [comment]
I will take care of this PR. thanks.

**Comment by toutdesuite** [comment]
Hi @sumens @lilin90 This PR has a problem: the Chinese version of the document need to be updated. Would you like to help translate other PR. Thanks a lot!

**Pull request merged**